src/EditorFeatures/Core/Implementation/CodeDefinitionWindow/DefinitionContextTracker.cs

```diff
@@ -0,0 +1,240 @@
+            _codeDefinitionWindowService = codeDefinitionWindowService;
+            _threadingContext = threadingContext;
+            _asyncListener = listenerProvider.GetListener(FeatureAttribute.CodeDefinitionWindow);
+        }
+
+        void ITextViewConnectionListener.SubjectBuffersConnected(ITextView textView, ConnectionReason reason, IReadOnlyCollection<ITextBuffer> subjectBuffers)
+        {
+            if (!_subscribedViews.Contains(textView) && !textView.Roles.Contains(PredefinedTextViewRoles.CodeDefinitionView))
+            {
+                _subscribedViews.Add(textView);
+                textView.Caret.PositionChanged += OnTextViewCaretPositionChanged;
+                QueueUpdateForCaretPosition(textView.Caret.Position);
+            }
+        }
+
+        void ITextViewConnectionListener.SubjectBuffersDisconnected(ITextView textView, ConnectionReason reason, IReadOnlyCollection<ITextBuffer> subjectBuffers)
+        {
+            if (reason == ConnectionReason.TextViewLifetime ||
+                !textView.BufferGraph.GetTextBuffers(b => b.ContentType.IsOfType(ContentTypeNames.RoslynContentType)).Any())
+            {
+                if (_subscribedViews.Contains(textView))
+                {
+                    _subscribedViews.Remove(textView);
+                    textView.Caret.PositionChanged -= OnTextViewCaretPositionChanged;
+                }
+            }
+        }
+
+        private void OnTextViewCaretPositionChanged(object? sender, CaretPositionChangedEventArgs e)
+        {
+            QueueUpdateForCaretPosition(e.NewPosition);
+        }
+
+        private void QueueUpdateForCaretPosition(CaretPosition caretPosition)
+        {
+            Contract.ThrowIfFalse(_threadingContext.JoinableTaskContext.IsOnMainThread);
+
+            // Cancel any pending update for this view
+            _currentUpdateCancellationToken?.Cancel();
+
+            // See if we moved somewhere else in a projection that we care about
+            var pointInRoslynSnapshot = caretPosition.Point.GetPoint(tb => tb.ContentType.IsOfType(ContentTypeNames.RoslynContentType), caretPosition.Affinity);
+            if (pointInRoslynSnapshot == null)
+            {
+                return;
+            }
+
+            _currentUpdateCancellationToken = new CancellationTokenSource();
+
+            var cancellationToken = _currentUpdateCancellationToken.Token;
+            var asyncToken = _asyncListener.BeginAsyncOperation(nameof(DefinitionContextTracker) + "." + nameof(QueueUpdateForCaretPosition));
+            UpdateForCaretPositionAsync(pointInRoslynSnapshot.Value, cancellationToken).CompletesAsyncOperation(asyncToken);
+        }
+
+        private async Task UpdateForCaretPositionAsync(SnapshotPoint pointInRoslynSnapshot, CancellationToken cancellationToken)
+        {
+            var locations = await GetContextFromPointAfterDelayAsync(pointInRoslynSnapshot, cancellationToken).ConfigureAwait(true);
```

Comment:
this needs a try/catch for cancellation and all exceptions.  Note: this is why i prefer 'async void' as it makes it clear that the result is not observed and we need to handle the fire-and-forget nature of it ourselves.

Reply:
I think we really need to do the other change mentioned elsewhere to add a "run this method under an async operation and also with handling" that I think we discussed privately. Willing to do that in a separate PR generally for the codebase.